### PR TITLE
Small tweaks to application setup

### PR DESCRIPTION
**Pull request by davidfowl**
- Close over WebApplicationBuilder for configuration and environment.
- Male explicit call to UseRouting so the error handler works
- Grouped middleware Use* before route Map*.

### CODE REVIEW

File: src/TodoApp/TodoApp.csproj [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <TypeScriptCompileBlocked>true</TypeScriptCompileBlocked>
     <TypeScriptToolsVersion>latest</TypeScriptToolsVersion>
     <UserSecretsId>TodoApp</UserSecretsId>
+    <Nullable>enable</Nullable>
```
Comment (martincostello) [inline src/TodoApp/TodoApp.csproj:9]: This is in the `Directory.Build.props` file already - is there some IDE tooling or something that doesn't like that being there?

File: src/TodoApp/AuthenticationModule.cs [hunk 2]
```diff
@@ -55,6 +55,7 @@ public static IServiceCollection AddGitHubAuthentication(this IServiceCollection
                     options.ClaimActions.MapJsonKey(GitHubAvatarClaim, "avatar_url");
                 }
             })
+            .ValidateOnStart()
```
Comment (martincostello) [inline src/TodoApp/AuthenticationModule.cs:58]: Nice.

**Review by martincostello — APPROVED** [review]

File: src/TodoApp/Program.cs [hunk 3]
```diff
@@ -27,15 +27,14 @@
...
-    builder.UseSqlite("Data Source=" + databaseFile);
+    options.UseSqlite("Data Source=" + databaseFile);
 });
 
 // Add user authentication with GitHub as an external OAuth provider
 builder.Services.AddGitHubAuthentication();
 
 // Add Razor Pages to render the UI
 builder.Services.AddRazorPages();
-builder.Services.AddRouting();
```
[the remark this answers is not in the document]
Comment (davidfowl) [reply src/TodoApp/Program.cs:38]: Yep this is implicit

File: src/TodoApp/TodoApp.csproj (hunk 1 above)
Comment (davidfowl) [reply src/TodoApp/TodoApp.csproj:9]: Missed that

Comment (davidfowl) [inline src/TodoApp/TodoApp.csproj:9]:
```suggestion
```

**Comment by martincostello** [comment]
Thanks!

File: src/TodoApp/ApiModule.cs [hunk 4]
```diff
@@ -52,7 +52,7 @@ public static IEndpointRouteBuilder MapTodoApiRoutes(this IEndpointRouteBuilder
             ITodoService service,
             CancellationToken cancellationToken) =>
             {
-                if (model is null || string.IsNullOrWhiteSpace(model.Text))
+                if (string.IsNullOrWhiteSpace(model.Text))
```
Comment (davidfowl) [inline src/TodoApp/ApiModule.cs:55]: This is new in rc1, we use the nullability of the parameter to determine if we should call your method with null. https://github.com/dotnet/aspnetcore/issues/32375

Comment (martincostello) [reply src/TodoApp/ApiModule.cs:55]: Lots of nice little QoL and usability stuff coming in now 😎

**Pull request merged**